sos/utilities.py

```diff
@@ -19,11 +19,26 @@
 import threading
 import time
 import io
-import magic
-
 from contextlib import closing
 from collections import deque
 
+# try loading magic>=0.4.20 which implements detect_from_filename method
+magic_mod = False
+try:
+    import magic
+    magic.detect_from_filename(__file__)
+    magic_mod = True
+except ImportError:
```

Comment:
We'd also need to catch `AttributeError` for cases where older `magic` is installed and `detect_from_filename()` is not present.